label_studio/ml_model_providers/serializers.py

```diff
@@ -29,3 +31,20 @@ def to_representation(self, instance):
         for attr in self.secure_fields:
             result.pop(attr)
         return result
+
+    def validate(self, data):
+        model_provider_connection = self.instance
+        if model_provider_connection:
+            for key, value in data.items():
+                setattr(model_provider_connection, key, value)
+        else:
+            model_provider_connection = self.Meta.model(**data)
+
+        try:
+            model_provider_connection.validate_api_key()
+        except AuthenticationError:
+            raise ValidationError("API key provided is not valid")
+        except NotImplementedError as e:
```

Comment:
Is it worth having a general `except Exception` here? I figure it would still bubble up as an error, but seems like it's probably helpful to have the context that it's a "validation error" in the sense that the error occurred while validating that we could instantiate an OpenAI client

Reply:
added